app/models/permission.rb

```diff
@@ -45,6 +61,8 @@ def validate_project_permission
         errors.add(:subject_id, 'cannot be blank if subject is present')
       elsif !Permission.valid_project_action?(action)
```

Comment:
It might be good to keep the permissions concept totally separate from the `Project` model, so I'm leaning toward not adding it there, but I do like your suggestion to update `Permission.valid_project_action?` to include the check.  I'll make that change.

Reply:
Actually, including the aggregator check inside `Permission.valid_project_action?` makes it a bit more complicated to add descriptive error messages, unless `Permission.valid_project_action?(action, project)` has a return value of `[valid, error_message]`.

Do you like this any better than the current code?
```
def self.valid_aggregator_project_action?(action)
    PROJECT_ACTIONS_DISALLOWED_FOR_AGGREGATOR_PROJECTS.include?(action)
end

def validate_project_permission
  if subject_id.blank?
    errors.add(:subject_id, 'cannot be blank if subject is present')
  else !Permission.valid_project_action?(action)
    errors.add(:action, "#{subject_id} is invalid")
  elsif !Project.find(subject_id).is_primary && !Permission.valid_aggregator_project_action?(action)
    errors.add(:action, "#{action} is not allowed for an aggregator project") unless 
  end
end
```

Reply:
It looks good. I honestly would have just used a generic "action isn't valid" error. But I liked your solution.